browsing-context.md

```diff
@@ -1,28 +1,295 @@
+
+### Restrictions on loaded content
+
+To simplify implementation, specification, and the web-developer facing consequences, prerendering browsing contexts cannot host non-HTTP(S) `Document`s. In particular, they cannot host:
+
+- `javascript:` URLs
+- `data:` URLs
+- `blob:` URLs
+- `about:` URLs, including `about:blank` and `about:srcdoc`
+
+In some cases, supporting these would create a novel situation for a top-level browsing context: for example, right now, a top-level browsing context cannot navigate to a `data:` or `blob:` URL, so allowing those to be prerendered and then activated (which is equivalent to a navigation) would require new implementation and specification infrastructure.
+
+In other cases, like `javascript:` URLs or `about:blank`, the problem is that those URLs generally inherit properties from their creator, and we don't want to allow this cross-`Document` influence for prerendered content. Overall, restricting to HTTP(S) URLs ensures that prerendered content always has a well-defined origin, that is not contingent on the referring page.
+
+The removal of the script-visible `about:blank` in prerendering browsing contexts also greatly simplifies them; its existence in other browsing contexts causes `Window`s and `Document`s to lose their normally one-to-one relationship.
+
+If a prerendering browsing context navigates itself to a non-HTTP(S) URL, e.g. via `window.location = "data:text/plain,foo"`, then the prerendering browsing context will be immediately discarded, and no longer be used by the user agent for anything.
 
 ## JavaScript API
 
-Script can observe and react to changes in the loading mode, if applicable. For example, it can observe a change from `uncredentialed-prerender` to `default` when a user navigates to the prerendered document. This can be used to defer personalization and logic which are not necessary for prerendering.
+We're still discussing various options for the API; please join that discussion in [#2](https://github.com/jeremyroman/alternate-loading-modes/issues/2). Here we present the considerations in play, plus our our tentative idea. The latter is mostly so that examples in this and other documents have something to exhibit.
+
+Goals:
+
+- Allow prerendered content to know that it's prerendered.
+- Allow prerendered content to know when it stops being prerendered.
+- Play nicely with existing or related APIs such as [the storage access API](https://github.com/privacycg/storage-access/) or the [page visibility API](https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API).
+
+Nice-to-haves:
+
+- Be general enough to accomodate other "alternate loading modes", such as [fenced frames](https://github.com/shivanigithub/fenced-frame/)
```

Comment:
Is life-cycle API or lifecycle state relevant here?

Reply:
Or do we rather want to make this a new API like loadingMode?  The mode currently mostly focuses on prerendering state, while the naming may imply that this can express other state related to loading, is there a possible extension that we wan to think about in a future?  E.g. what about loaded from prefetch cache, or bfcache?

Reply:
Very interesting. We haven't discussed freezing in this explainer at all. Perhaps it's a good idea to add a separate section. I suspect we'd just use the existing page lifecycle API though, and not integrate with the loadingMode one.

Whether we want a new API (vs. using existing APIs like visibilityState), and what the scope of that API is, are still open questions. I've tried to link to #2 to discuss that.